### PR TITLE
Support for extra parameters associated with nested proofs

### PR DESCRIPTION
**Pull request by dtebbs**
- Restructuring remaining structs to better support this
- NestedTransactions carry parameters as a hex string, returned from the aggregator as part of each batch.
- Tests to use the parameter in dummy_app and check the value.

Original goal was to support arbitrary structures per app, encoded into bytes which get decoded by the app. After wrestling with solidity and web3.py I hit problems passing `bytes[batch_size]` to contracts, either directly or encoded with `eth_abi.encode` (the corresponding decode on the contract seems to fail).  So for now the nested parameters are still passed as `uint256[]`.

### CODE REVIEW

File: client/zecale/cli/zecale_submit.py [hunk 1]
```diff
@@ -2,23 +2,20 @@
 #
 # SPDX-License-Identifier: LGPL-3.0+
 
-from .utils import load_extended_proof
-from ..core.nested_transaction import NestedTransaction
-from click import option, command, pass_context, Context
+from .utils import load_nested_transaction
```
Comment (AntoineRondelet) [inline client/zecale/cli/zecale_submit.py:5]: What is the convention you follow for imports syntax? I can see that absolute paths are used as in `client/zecale/cli/utils.py` for instance, but here you use relative imports. Why not stick to absolute imports everywhere (the project is not very deep - at least for now..)

File: libzecale/core/nested_transaction.hpp [hunk 2]
```diff
@@ -19,25 +19,29 @@ template<typename nppT, typename nsnarkT> class nested_transaction
 private:
     std::string _application_name;
     std::shared_ptr<libzeth::extended_proof<nppT, nsnarkT>> _extended_proof;
+    std::vector<uint8_t> _parameters;
```
Comment (AntoineRondelet) [inline libzecale/core/nested_transaction.hpp:22]: What prevents malicious manipulation of these `_parameters` by the aggregator? i.e. what proves that `nested_parameters` in the `AggregatedTransaction` proto message is honestly computed as the concatenation of all `parameters` from the popped nested transactions processed as part of the last batch?

File: client/zecale/cli/zecale_submit.py (hunk 1 above)
Comment (dtebbs) [reply client/zecale/cli/zecale_submit.py]: I find it easier to deal with for groups of related files (e.g. groups of commands etc).  But no problem, it's trivial and actually got switched to absolute during the rebase.

Comment (AntoineRondelet) [reply client/zecale/cli/zecale_submit.py]: Thanks

File: scripts/test_client [hunk 3]
```diff
@@ -57,25 +64,19 @@ pushd user1
...
 
     # Get a single batch and send to the contract
     zecale get-batch --name ${APP_NAME} --batch-file batch1.json
     zecale check-batch batch1.json --batch-size 2
     zecale submit-batch batch1.json --wait
 
     echo Check app state after batch 1
-    zecale_dummy_app get 7 --check 1
-    zecale_dummy_app get 8 --check 1
+    zecale_dummy_app get 7 --check 23
```
Comment (AntoineRondelet) [inline scripts/test_client:78]: Hum.. isn't the value of the flag `--check` supposed to be {0,1}? (see the flag documentation pasted below)
```python
"--check", type=int, help="Check the result against the given value (0|1)")
```

Comment (dtebbs) [reply scripts/test_client:78]: I've updated the help text.
(The type as been switched to `int`, so it no longer needs to be `0|1`.  This is how we check that the extra parameters for each nested tx get propagated to the client application.)

**Review by AntoineRondelet — APPROVED** [review]
LGTM

**Pull request merged**